pkg/manifest/subscription.go

```diff
@@ -138,6 +142,13 @@ func subscriptionService(subscriptionOptions subscription.ImageOptions, serviceO
 			commands = append(commands, "/usr/bin/insights-client --register")
 			// insights-client creates the .gnupg directory during boot process, and is labeled incorrectly
 			commands = append(commands, "restorecon -R /root/.gnupg")
+			// register to template if template is specified. For Prod only.
+			if subscriptionOptions.TemplateUUID != "" {
+				patchURL := fmt.Sprintf("%s/api/patch/v3/templates/%s/subscribed-systems", "https://cert.console.redhat.com", subscriptionOptions.TemplateUUID)
+				curlToAssociateSystem := fmt.Sprintf("curl -v --retry --proxy %s %s --cert /etc/pki/consumer/cert.pem  --key /etc/pki/consumer/key.pem -X PATCH", subscriptionOptions.Proxy, patchURL)
```

Comment:
I think that it would be desirable to make this deployment-agnostic if possible and not mention prod at all. Would it make sense to move the URL to the `subscription.ImageOptions` and have it be specified by the library caller, instead of hardcoding it in the library itself? I have the same question for the API endpoint template, but I expect that to be already deployment-agnostic and stable.

Reply:
I think that makes sense. I pushed a change to that effect. If you had something else in mind I'm happy make other changes.